### PR TITLE
feat: missing variable improvement - [INS-4396]

### PR DESCRIPTION
**Pull request by CurryYangxx**
**Changes**

show all the missing environment variables in request
image
image

### CODE REVIEW

**Comment by jackkav** [comment]
On a related note, in order to feedback useful context to a CLI user, we may want to change how we handle these. Theres a ticket to address it here INS-4142, provided this PR doesn't make that any more difficult it should be fine.

**Comment by CurryYangxx** [comment]
> On a related note, in order to feedback useful context to a CLI user, we may want to change how we handle these. Theres a ticket to address it here [INS-4142](https://linear.app/insomnia/issue/INS-4142), provided this PR doesn't make that any more difficult it should be fine.

@jackkav  This PR just get all missing variable at once and change some UI, will not affect the original error handling. So I think this will not let it more difficult.

File: packages/insomnia/src/templating/utils.ts [hunk 1]
```diff
@@ -280,13 +281,24 @@ export function decodeEncoding<T>(value: T) {
...
+    let variable = match[1];
+    if (variable.includes('_.')) {
+      variable = variable.split('_.')[1];
+    }
+    // Check if the variable is not present in the context
+    if (_.get(templatingContext, variable) === undefined) {
+      missingVariables.push(variable);
+    }
+  }
+  return missingVariables;
```
Comment (CurryYangxx) [inline packages/insomnia/src/templating/utils.ts]: compare with template context and get all empty variables in one editor.
image

**Review by ihexxa — APPROVED** [review]

**Comment by ihexxa** [comment]
image
Checked and did some tests, looks good.

**Pull request merged**